src/Build/Evaluation/Evaluator.cs

```diff
@@ -2370,12 +2370,15 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
 
             if (importElement.ParsedSdkReference != null)
             {
-                // Try to get the solution path when available.
+                // Try to get the path to the solution and project being built. If the solution path equals "*Undefined*" 
+                // (hard coded in Microsoft.Common.CurrentVersion.targets) set to null for consistency.
```

Comment:
Instead of 

> If the solution path equals "*Undefined*" (hard coded in Microsoft.Common.CurrentVersion.targets) set to null for consistency.

Maybe

> The solution path is not directly known in MSBuild. It is passed in as a property either by the VS project system or by MSBuild's solution metaproject. Microsoft.Common.CurrentVersion.targets sets the value to `*Undefined*` when not set, and for backward compatibility, we shouldn't change that. But resolvers should be exposed to a string that's null or a full path, so correct that here.